src_py/grndwork_api_client/make_request.py

```diff
@@ -0,0 +1,99 @@
+import json
+from typing import Any, Dict, Iterator, Optional, Tuple
+
+import requests
+from requests.exceptions import RequestException
+
+from .dtos import ContentRange
+
+
+def make_request(
+    url: str,
+    *,
+    method: str,
+    token: str,
+    headers: Optional[Dict[str, Any]] = None,
+    query: Any = None,
+    body: Any = None,
+) -> Tuple[Any, ContentRange]:
+
+    if not headers:
+        headers = {}
+
+    if token:
+        headers['Authorization'] = f'Bearer {token}'
+
+    if body:
+        headers['Content-Type'] = 'application/json'
+
+    try:
+        response = requests.request(
+            method=method,
+            url=url,
+            headers=headers,
+            params=query,
+            data=json.dumps(body),
+        )
+        cont_range = parse_content_range(response.headers['Content-Range'])
```

Comment:
Only parse content range if included in response header

Reply:
See comment below.